client/src/components/dashboard-component/index.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+import { Typography, Container } from "@mui/material";
+import InputAdornment from "@mui/material/InputAdornment";
+import SearchIcon from "@mui/icons-material/Search";
+import DashboardHeaderComponent from "./header-dashboard-component";
+import mockData from "./mockData";
+import {
+  MainContainer,
+  SearchFieldContainer,
+  SearchField,
+  Divider,
+} from "./index.styles";
+import DashboardNavigationComponent from "./nav-dashboard-component";
+import UpdatesCollection from "./update-collection-component";
+
+function DashboardComponent() {
+  const firstName = "First Name";
+
+  const mockJobUpdates = mockData;
+
+  const [jobUpdates] = useState(mockJobUpdates);
+
+  return (
+    <Container>
```

Comment:
This might be better to have built out in the `pages` directory. Especially once it gets hooked up to routing and the backend all the requests can be made from a "page" instead of a "component"
This is just my thought, so I wanted to propose the idea and see what you think.

Reply:
@jordanjanakievski  Hi Jordan, yes I agree. I just added a useState temporarily directly in the component just for mock data purposes but my thoughts on where to build out the page is by using redux store.

So I was thinking we set up the react routes and everything within the App.jsx file and we wrap the entire app by the redux component so for example 

`
function App() {
  const isAuthenticated = true;
  var redirect = "/signin";

  return (
    <Provider store={store}>
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              isAuthenticated ? (
                <Navigate to="/home" />
              ) : (
                <Navigate to="/signin" />
              )
            }
          />
          <Route path="/login" element={<LoginPage />} />
          <Route path="/signin" element={<SignInPage />} />
          <Route path="/signup" element={<SignUpPage />} />
          ........ and so on
`
where each element is under the pages directory that I have set up. And the provider is the redux component with our store that has all the data we want to keep stored and processed from the backend. So for example when we route to the dashboard, the dashboard page under the page directory is like an intermediate step where we collect all needed data (we will have a loading mask or something) and then once we get all the data we pass it into our dashboard component. But sometimes actions within the dashboard (such as creating a new job lead etc), the dashboard component will have actions that will manipulate the data, that's why I was thinking we should have a centralized redux store where all components can access at any time instead of always passing props within.

Reply:
This sounds like a good idea. Once Daniel gets a chance to take a look we can definitely discuss it. We can then update the team at the meeting after reading week. @DanielDervishi what do you think?
Thank you!

Reply:
Hey @jordanjanakievski and @kevinle623. I've never used redux before but after doing some research it seems pretty cool and definitely something that we could use. I like the fact that state doesn't have to be passed down the component tree.

Although given that we would sort of pre-load data before the dashboard is actually loaded, my question is wouldn't this be really resource intensive? Like if we had to load ALL the application's data at once? Also what happens when a component is no longer present, does its data persist throughout the lifecycle of the application? This doesn't seem ideal memory wise, so some clarification on this would be great.

Also I'm not sure I totally understand the example when it comes to the dashboard actions manipulating data. It would be great if you could further explain that. If I did understand correctly, what's the advantage of modifying the data in the redux store directly over just fetching from the database again to get the updated data and updating the state.

Thanks!

Reply:
Hi @DanielDervishi and @jordanjanakievski. In terms of resource intensive and the idea of loading all the application's data at once isn't really the case, redux has something called slices. So we have a full redux store that has all our data that we would want to have in our app, but we can set up individual slices depending on the "slice" of the state that we want to maybe populate or store for a specific component if that makes sense. So maybe we would have a slice specifically for job leads, another slice specifically for employees. So with this, maybe an example would be us loading into a page where we would only need job leads, so then we would call the backend to only get job leads data and then we store it into our redux but only specifically the job leads.  So we don't need to store everything all at once, just whenever we need it.

So in regards to a component not being present in the dom anymore, the redux store will still have data persisted depending on whatever we give or provide to it, but anything that the component holds locally will be gone.

For the example of dashboard actions manipulating data. What I meant was within the dashboard we can add a new job leads, employees etc. so for example in our redux store we would have a list of employees or job leads, so an action could be adding a new job lead which will cause us to add to the redux store, then we can have an operation to sync it / add to the backend database as well. 

For your final question about the advantage, it would be more so reducing the amount of times we have to fetch from the database until we absolutely need to. So back to the example of job leads, lets say we load into a page that in primarly storing job leads so we would fetch data from that database for that. but then maybe there's another page that uses the same exact data such as an overview page of job leads and employees and this page is far from the job leads page in terms of the react tree so if we store it in our redux store as well then we can get that data from there.

So in general, the main advantage of having a redux store is as follows

1. easy for components that maybe "far away" from each other to still keep a centralilzed source of truth with consistent data
2. redux store prevents unneccessary re-renders to components as well since only components that are relevant to a changed data will rerender and others that don't use that specific data will stay the same (performance). Because if we passing everything through props, any small change to a parent components will force a rerender on ALL the child components.

But all in all, this was more so a suggestion and for situations like this project I was just thinking about using redux. But just like how I suggested styled-components, I feel like this can just be something to consider in the future if we run into troubles so might be best for now to not use redux state and try to get everything orchestrated through vanilla react builtin useStates and passing props.

Reply:
Hi Kevin, we think the state management stuff sounds great but do not need it currently in the project. There could definitely come a time in these next couple of months when it could become a necessity and we will for sure look to implement Redux. On Monday, Daniel and I will have another update regarding the next sprint and this could become a talking point for one or two of the tickets. We really appreciate these explanations and that you are sharing your experience and knowledge with us to get the best outcome for this project.